### ISSUE TITLE
[Enhancement] Add sun level emulation options for Boktai titles

### ISSUE THREAD

**Issue by Dartz150** [Gericom/GBARunner3#28]
Botkai titles have an UV sensor for gameplay purposes as mentioned [here](https://boktai.neoseeker.com/wiki/Solar_Sensor)

Related titles:

Boktai: The Sun Is in Your Hand.
Boktai 2: Solar Boy Django.
Boktai 3: Sabata's Counterattack.

**Comment by LanHikariDS** [comment]
I wonder, would it be possible to pass GPIO information from a real cartridge? That way one could play the English translation patch for Boktai 3 with an official sensor.

I can't imagine it'll be _too_ helpful, since you seem to already understand GPIO stuff enough to implement Rumble Pak support in an experimental build, but Lunar Knights supports reading sensor info from a Slot 2 cart as well, and _seems_ to support all regions of the original games (Tested and confirmed with Boktai US/JP, Boktai 2 US, and a flash cart with Boktai 3 JP, and the US Lunar Knights identified all of them correctly, even showing the Japanese Boktai 1 logo when that region was inserted)

**Comment by Dartz150** [comment]
> I wonder, would it be possible to pass GPIO information from a real cartridge? That way one could play the English translation patch for Boktai 3 with an official sensor.
> 
> I can't imagine it'll be _too_ helpful, since you seem to already understand GPIO stuff enough to implement Rumble Pak support in an experimental build, but Lunar Knights supports reading sensor info from a Slot 2 cart as well, and _seems_ to support all regions of the original games (Tested and confirmed with Boktai US/JP, Boktai 2 US, and a flash cart with Boktai 3 JP, and the US Lunar Knights identified all of them correctly, even showing the Japanese Boktai 1 logo when that region was inserted)

Yes, it's possible, we would just need to read directly from the Slot-2 memory area, and any cart with a Solar Sensor should work fine. This could be implemented once DSi camera support (and maybe key combinations for non-DSi models) as the solar sensor is added.
